engines/sci/engine/message.cpp

```diff
@@ -183,10 +184,17 @@ class MessageReaderV4_MacSCI32 : public MessageReader {
 #endif
 
 bool MessageState::getRecord(CursorStack &stack, bool recurse, MessageRecord &record) {
-	Resource *res = g_sci->getResMan()->findResource(ResourceId(kResourceTypeMessage, stack.getModule()), false);
+	// find a workaround for the requested message and use the prescribed module
+	int module = stack.getModule();
+	MessageTuple &tuple = stack.top();
```

Comment:
Why don't you keep using the MessageTuple wherever you're passing the (noun, verb, cond, seq) tuple?
It would make for clearer code IMO.

Reply:
If you're asking why findMessageWorkaround() takes noun/verb/cond/seq parameters as opposed to just a MessageTuple, that would indeed be clearer, and that's how I first wrote it, but then workarounds.cpp has to include message.h and I didn't want to introduce that since I'm now including workarounds.h from message.cpp. It would compile, but I didn't like it, so I punted and went with the individual parameters.

Thanks for checking it out! You have absolutely found the line I grumbled about the most.

Reply:
You can always move MessageTuple to another file, but it's not a big deal. It's only a readability (and maintainability) issue. I only noticed this because the noun/verb/cond/seq parameters are all over the codebase which is a sign of high cohesion. A future PR perhaps.